utils/check-hardware-input.ts

```diff
@@ -29,12 +29,10 @@ export function parsingCpuType(cpuType: string): Cpu {
  * @param memoryType The memory type as a string.
  * @returns The corresponding Memory enum value.
  */
-export function parsingMemoryType(memoryType: string): Memory {
+export function parseMemoryType(memoryType: string): Memory {
     switch (memoryType) {
         case 'TWO_GB':
             return Memory.TWO_GB;
```

Comment:
**Suggestion:** Consider adding a default case to handle unexpected `memoryType` values in the `parseMemoryType` function, which will help in debugging and provide a fallback mechanism.
> 🍄  **Medium Priority** 
 > [error handling, importance: 7]
```suggestion
    switch (memoryType) {
        case 'TWO_GB':
            return Memory.TWO_GB;
        // other cases...
        default:
            throw new Error(`Unexpected Memory type: ${memoryType}`);
    }
```